core/src/apps/cardano/shelley/utils.py

```diff
@@ -0,0 +1,13 @@
+# https://en.wikipedia.org/wiki/Variable-length_quantity
+def variable_length_encode(i):
+    encoded = []
+
+    bit_length = len(bin(i)[2:])
+    encoded.append(i & 127)
+
+    while bit_length > 7:
+        i >>= 7
+        bit_length -= 7
+        encoded.insert(0, (i & 127) + 128)
+
+    return bytes(encoded)
```

Comment:
This should throw if `i` is negative. Check the existing byron code for error reporting, it seems we use `wire.ProcessError` at the top level. 

Perhaps it would be better to raise ordinary python errors (e.g. ValueError), and catch those and convert them into wire errors at the top level (where processing protobufs) instead of throwing wire errors from all over the place.